java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -658,6 +671,10 @@ private boolean isAccessible(Env env, JavaSymbol.TypeJavaSymbol site, JavaSymbol
     }
   }
 
+  static boolean sameOutermostClass(JavaSymbol s1, JavaSymbol s2) {
+    return s1.outermostClass() == s2.outermostClass();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)